user_management/models/mixins.py

```diff
@@ -128,56 +132,48 @@ def create_superuser(self, email, password, **extra_fields):
+    """Define how validation and password reset emails are sent.
+
+    `password_reset_notification` and `validation_notification` can be overriden to
+    provide custom settings to send emails.
+    """
+    password_reset_notification = PasswordResetNotification
+    validation_notification = ValidationNotification
 
     def email_context(self, site):
         return {
+            'protocol': 'https',
             'uid': urlsafe_base64_encode(force_bytes(self.pk)),
             'token': default_token_generator.make_token(self),
             'site': site,
         }
 
-    def email_kwargs(self, context, domain):
-        """Prepare the kwargs to be passed to incuna_mail.send"""
-        return {
-            'to': [self.email],
-            'template_name': self.TEXT_EMAIL_TEMPLATE,
-            'html_template_name': self.HTML_EMAIL_TEMPLATE,
-            'subject': self.get_email_subject(domain),
-            'context': context,
-        }
-
-    def get_email_subject(self, domain):
-        return _(self.EMAIL_SUBJECT).format(domain=domain)
-
     def send_validation_email(self):
-        """
-        Send a validation email to the user's email address.
-
-        The email subject can be customised by overriding
-        VerifyEmailMixin.EMAIL_SUBJECT or VerifyEmailMixin.get_email_subject.
-        To include your site's domain in the subject, include {domain} in
-        VerifyEmailMixin.EMAIL_SUBJECT.
-
-        By default send_validation_email sends a multipart email using
-        VerifyEmailMixin.TEXT_EMAIL_TEMPLATE and
-        VerifyEmailMixin.HTML_EMAIL_TEMPLATE. To send a text-only email
-        set VerifyEmailMixin.HTML_EMAIL_TEMPLATE to None.
-
-        You can also customise the context available in the email templates
-        by extending VerifyEmailMixin.email_context.
-
-        If you want more control over the sending of the email you can
-        extend VerifyEmailMixin.email_kwargs.
-        """
+        """Send a validation email to the user's email address."""
         if not self.email_verification_required:
             raise ValueError(_('Cannot validate already active user.'))
 
         site = Site.objects.get_current()
-        context = self.email_context(site)
-        send(**self.email_kwargs(context, site.domain))
+        email_subject = _('{domain} account validate'.format(domain=site.domain))
+
+        notification = self.validation_notification(
+            user=self,
+            email_subject=email_subject,
+            context=self.email_context(site),
+        )
+        notification.notify()
+
+    def send_password_reset(self):
+        """Send a password reset to the user's email address."""
+        site = Site.objects.get_current()
+        email_subject = _('{domain} password reset'.format(domain=site.domain))
+
+        notification = self.password_reset_notification(
+            user=self,
+            email_subject=email_subject,
+            context=self.email_context(site),
```

Comment:
I don't see why the notification should not determine the context. It's already defining the templates. What if someone wants to do this over SMS or pigeon? Should we not allow them to do it? Is that not the point of `incuna-pigeon`
?